go/vt/discovery/fake_healthcheck.go

```diff
@@ -74,7 +77,30 @@ func (fhc *FakeHealthCheck) RegisterStats() {
 
 // WaitForAllServingTablets is not implemented.
 func (fhc *FakeHealthCheck) WaitForAllServingTablets(ctx context.Context, targets []*querypb.Target) error {
-	return nil
+	if !fhc.ShouldWaitForTablets {
```

Comment:
all the other tests rely on the fake implementation. We could update them in a different PR to simplify this. For now added this extra option.